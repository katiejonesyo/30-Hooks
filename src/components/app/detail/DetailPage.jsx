import React, { useState, useEffect } from 'react';
import { useParams } from 'react-router-dom';
import { getCharactersById } from '../../../services/AvatarApi';
import { Link } from 'react-router-dom';

export default function DetailPage() {
  const [character, setCharacter] = useState({});
  const { id } = useParams();
  console.log(id, 'character id');

  useEffect(() => {
    getCharactersById(id).then((character) => setCharacter(character));
  }, []);

  const { name, allies, enemies, photoUrl } = character;

  return (
    <div>
      <h1>{name}</h1>
      <img src={photoUrl} />
      <ul>
        <li>Allies: {allies}</li>
        <li>Enemies: {enemies}</li>
      </ul>
      <Link to="/">List Page</Link>
    </div>
  );
}


