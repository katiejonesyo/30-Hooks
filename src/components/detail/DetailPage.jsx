import React, { useState, useEffect } from 'react';
import { useParams } from 'react-router-dom';
import { getCharactersById } from '../../services/AvatarApi';
import { Link } from 'react-router-dom';

export default function DetailPage() {
  const [character, setCharacter] = useState({});
  const { id } = useParams();

  useEffect(() => {
    getCharactersById(id).then((character) => setCharacter(character));
  }, []);

  const { name, allies, enemies, photoUrl } = character;

  return (
    <>
    <div>
      <h4>To return to the List Page just click the link under the charcter photo on the detail page.</h4>
      <h1>{name}</h1>
      <img src={photoUrl} />
      <ul>
        <li>Allies: {allies}</li>
        <li>Enemies: {enemies}</li>
      </ul>
      <Link to="/">List Page</Link>
    </div>
    </>
  );
}


