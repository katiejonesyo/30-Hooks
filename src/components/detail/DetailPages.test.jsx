import React from 'react';
import { render, cleanup } from '@testing-library/react';
import DetailPage from './DetailPage';
import { MemoryRouter } from 'react-router-dom';

describe('DetailPage component', () => {
  afterEach(() => cleanup());
  it('renders a details of a single character', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <DetailPage />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});


