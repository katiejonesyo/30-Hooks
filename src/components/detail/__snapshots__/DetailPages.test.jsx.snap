// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailPage component renders a details of a single character 1`] = `
<DocumentFragment>
  <div>
    <h4>
      To return to the List Page just click the link under the charcter photo on the detail page.
    </h4>
    <h1 />
    <img />
    <ul>
      <li>
        Allies: 
      </li>
      <li>
        Enemies: 
      </li>
    </ul>
    <a
      href="/"
    >
      List Page
    </a>
  </div>
</DocumentFragment>
`;
